staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/promise/interface.go

```diff
@@ -52,78 +58,10 @@ type LockingReadable interface {
 	// any number of times.  IsSet may acquire, and later release, the
 	// lock any number of times.
 	IsSetLocked() bool
-}
-
-// WriteOnceOnly represents a variable that is initially not set and
-// can be set once.
-type WriteOnceOnly interface {
```

Comment:
In case anyone is not aware, the Go language has improved in a relevant way here since these interfaces were defined.  In the bad old days, Go did not allow a diamond-shaped inheritance graph among interfaces.  That is why this is written in mixin-style.  Now, Go supports diamon-shaped inheritance.  It would be possible to maintain the producer/consumer distinction here with fewer interface definitions.

Reply:
I think we shouldn't be introducing interfaces for the sake of introducing them. Let's not overcomplicate stuff and not introduce things that we don't even use/need anywhere.